llvm/lib/ExecutionEngine/JITLink/aarch32.cpp

```diff
@@ -377,12 +377,14 @@ Expected<int64_t> readAddendThumb(LinkGraph &G, Block &B, const Edge &E,
                   : decodeImmBT4BlT1BlxT2(R.Hi, R.Lo);
 
   case Thumb_MovwAbsNC:
+  case Thumb_MovwPrelNC:
```

Comment:
We were talking about the opcode checks and duplication in both, code and constants. I've been working on a patch to compensate for it here: https://github.com/llvm/llvm-project/pull/71649 With that the asymmetry in opcode checks will go away. I think we can land this as is.